tiledb/sm/query/query.cc

```diff
@@ -361,26 +363,32 @@ Status Query::capnp(rest::capnp::Query::Builder* queryBuilder) const {
           return Status::RestError(
               "Cannot serialize query; variable-length attribute has no offset "
               "buffer set.");
-        // Var-length data
-        size_t buff_length =
-            (*buff.buffer_var_size_) / datatype_size(attr_type);
-        auto builder = attributeBuffer.initBuffer();
-        RETURN_NOT_OK(rest::capnp::utils::set_capnp_array_ptr(
-            builder, attr_type, buff.buffer_var_, buff_length));
-        // Offsets
-        attributeBuffer.setBufferOffset(kj::arrayPtr(
-            static_cast<const uint64_t*>(buff.buffer_),
-            (*buff.buffer_size_) / sizeof(uint64_t)));
+        totalNumOfBytesInVarBuffers += *buff.buffer_var_size_;
+        attributeBufferHeader.setBufferSizeInBytes(*buff.buffer_var_size_);
+        attributeBufferHeader.setDatatypeSizeInBytes(datatype_size(attr_type));
+        totalNumOfBytesInOffsets += *buff.buffer_size_;
+        attributeBufferHeader.setOffsetBufferSizeInBytes(*buff.buffer_size_);
+        RETURN_NOT_OK(set_buffer(
```

Comment:
I don't think this call to `set_buffer()` is required, is it?

Reply:
https://github.com/Shelnutt2/TileDB/blob/2eec9e4aaaa022d968e660d83b56804b8d79c7aa/tiledb/rest/capnp/query.cc#L55 calls `capnp`, we iterate through attributes and for each one we have to call `set_buffer` in order for https://github.com/Shelnutt2/TileDB/blob/2eec9e4aaaa022d968e660d83b56804b8d79c7aa/tiledb/rest/capnp/query.cc#L108 `get_buffer` to find later the buffer set, i think it is critical to the logic